### DISCUSSION TITLE
Why does blend_images take so long to run?

### DISCUSSION THREAD

**Discussion by bakermanbrian** [OpenStitching/stitching#179] [Q&A]
I've been running this project and it's very fast (thanks!) except for this one part:

```
    def blend_images(self, imgs, masks, corners):
        for idx, (img, mask, corner) in enumerate(zip(imgs, masks, corners)):
            if self.timelapser.do_timelapse:
                self.timelapser.process_and_save_frame(
                    self.images.names[idx], img, corner
                )
            else:
                self.blender.feed(img, mask, corner)
```

I have self.timelapser.do_timelapse as False and I have done time tests and the self.blender.feed(img, mask, corner) line is quite fast. It seems that the loop iteration is the slow part.

So even this is slow for me:

```
    def blend_images(self, imgs, masks, corners):
        for idx, (img, mask, corner) in enumerate(zip(imgs, masks, corners)):
            pass
```

How would I go about speeding up this loop iteration?

**Comment by lukasalexanderweber** [comment]
this is because stitching uses generators to optimize memory usage. Whenever yield is used, a generator is used. Elements are only loaded when they are actually needed. This is the case in the corresponding `blend_images` function at the end of the pipeline, even if nothing is done with the data (`pass`). To see which part of the stitching takes the longest, it would probably be better to measure the times in [verbose stitching](https://github.com/OpenStitching/stitching/blob/main/stitching/verbose.py). whenever list() is called, a generator is converted to a list, which means that the data is actually loaded. If you are interested in the topic in more detail, you can also take a look at the [performance comarison test](https://github.com/OpenStitching/stitching/blob/main/tests/test_performance.py).
